packages/integration-karma/test/light-dom/events/index.spec.js

```diff
@@ -0,0 +1,221 @@
+        nodes = createTestElement('x-light-child', LightChild);
+    });
+    afterEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+    });
+    it('{ bubbles: true, composed: false }', () => {
+        const log = dispatchEventWithLog(
+            nodes.button,
+            nodes,
+            new CustomEvent('test', { bubbles: true, composed: false })
+        );
+
+        const composedPath = [
+            nodes.button,
+            nodes['x-light-child'],
+            document.body,
+            document.documentElement,
+            document,
+            window,
+        ];
+
+        expect(log).toEqual([
+            [nodes.button, nodes.button, composedPath],
+            [nodes['x-light-child'], nodes.button, composedPath],
+            [document.body, nodes.button, composedPath],
+            [document.documentElement, nodes.button, composedPath],
+            [document, nodes.button, composedPath],
+            [window, nodes.button, composedPath],
+        ]);
+    });
+
+    it('{ bubbles: false, composed: false }', () => {
+        const log = dispatchEventWithLog(
+            nodes.button,
+            nodes,
+            new CustomEvent('test', { bubbles: false, composed: false })
+        );
+
+        const composedPath = [
+            nodes.button,
+            nodes['x-light-child'],
+            document.body,
+            document.documentElement,
+            document,
+            window,
+        ];
+
+        expect(log).toEqual([[nodes.button, nodes.button, composedPath]]);
+    });
+});
+
+describe('shadow container, light child', () => {
+    let nodes;
+    beforeEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', true);
+        nodes = createTestElement('x-shadow-container', ShadowContainer);
+    });
+    afterEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+    });
+
+    it('{ bubbles: true, composed: true }', () => {
+        const log = dispatchEventWithLog(
+            nodes.button,
+            nodes,
+            new CustomEvent('test', { bubbles: true, composed: true })
+        );
+
+        const composedPath = [
+            nodes.button,
+            nodes['x-light-child'],
+            nodes['x-shadow-container'].shadowRoot,
+            nodes['x-shadow-container'],
+            document.body,
+            document.documentElement,
+            document,
+            window,
+        ];
+
+        expect(log).toEqual([
+            [nodes.button, nodes.button, composedPath],
+            [nodes['x-light-child'], nodes.button, composedPath],
+            [nodes['x-shadow-container'].shadowRoot, nodes.button, composedPath],
+            [nodes['x-shadow-container'], nodes['x-shadow-container'], composedPath],
+            [
+                document.body,
+                process.env.NATIVE_SHADOW ? nodes['x-shadow-container'] : nodes.button,
```

Comment:
Please reference the related github issue for this discrepancy between native and synthetic.

Reply:
Discussed this with @abdulsattar. I don't think this behavior is what we want since this "workaround" was originally put into place to avoid breaking existing code. I don't think we should ship light dom components before plugging this leak where events originating from light dom components are not retargeted when handled by listeners on document or document.body.